### PR TITLE
ConsoleReader

### PR DESCRIPTION
**Pull request by Stanislava88**
Task for reading numbers from the console

### CODE REVIEW

File: src/console/ConsoleNum.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package console;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class ConsoleNum {
```
Comment (mgenov) [inline src/console/ConsoleNum.java]: Name of the class cannot be same as name of the method, when classes are nouns and methods are verbs.

**Comment by Stanislava88** [comment]
I changes the names

**Comment by Stanislava88** [comment]
@mgenov

File: src/console/Number.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Scanner;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class Number {
+  public boolean read() throws OutsideOfIntervalException {
+    Scanner input = new Scanner(System.in);
+    int number = input.nextInt();
+    if (number > 0 && number < 100) {
```
Comment (mgenov) [inline src/console/Number.java]: Could you make it to work and for 50 to 200 ?

**Comment by Stanislava88** [comment]
I make interval between n and m

File: src/list/ListOfObject.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package list;
+
+
+import java.util.Arrays;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class ListOfObject {
+  private Object[] list = {"Maria", 95};
```
Comment (mgenov) [inline src/list/ListOfObject.java]: You have to initialise array with some initial size and you don't have to copy array in the Add method. Now you are performing too many use-less operations. 

You can keep index to the current element and increment it when you add new element or decrement it when you remove one.

**Comment by Stanislava88** [comment]
done

File: src/list/ListOfObject.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  private Object[] list = new Object[size];
+  public int i=0;
+
+  public void add(Object obj) {
+    if(i<list.length)
+      list[i] =obj;
+    i++;
+  }
+
+  public void remove() {
```
Comment (mgenov) [inline src/list/ListOfObject.java]: I'm not sure that this is working, cause on adding you are adding at the beginning of the array, whether in remove, you are removing only the last element.

File: src/list/ListOfObject.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    if (list[list.length - 1] != null) {
+      list[list.length - 1] = null;
+      System.out.println(Arrays.toString(list));
```
Comment (mgenov) [inline src/list/ListOfObject.java]: This System.out need to be removed.

**Comment by Stanislava88** [comment]
Fixed remove method and removed System.out

File: src/list/ListOfObject.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  public int i = 0;
+
+  public void add(Object obj) {
+    if (i < list.length) {
+      list[i] = obj;
+      i++;
+    }
+  }
+
+  public void remove() {
```
Comment (mgenov) [inline src/list/ListOfObject.java]: What will happen when I execute: 

```java
ListOfObject list = new ListOfObject(); 
list.remove(); 
```

?

**Comment by Stanislava88** [comment]
Will occur exception

**Comment by Stanislava88** [comment]
Exception will occur

File: src/list/ListOfObject.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+  public void add(Object obj) throws FullListException {
+    if (i < list.length) {
+      list[i] = obj;
+      i++;
+    } else throw new FullListException("The list is full.You can not add element!!!");
+  }
+
+  public void remove() throws EmptyListException {
+    if (list[i] == null) {
```
Comment (mgenov) [inline src/list/ListOfObject.java]: I see that you are relaying on the fact that current element is null. What if null element is added? 

add(null); 
remove(); 

Using of nulls for such things is not a good idea. 

```java
// no elements are added 
if (i== 0) {
   return; // or throw exception, but it's not a good idea 
}
list[i - 1] = null; 
i--; 
```

Comment (mgenov) [inline src/list/ListOfObject.java]: As this is exception task, you can keep throwing of exception.

**Comment by Stanislava88** [comment]
Added exception for null value

**Comment by mgenov** [comment]
Not a good solution for production code, but for learning of exceptions seems fine. 

LGTM !

**Pull request merged**